runtime/src/main/java/org/corfudb/runtime/LogReplicationServiceClient.java

```diff
@@ -0,0 +1,339 @@
+import org.corfudb.runtime.collections.TxnContext;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuInterruptedError;
+import org.corfudb.util.retry.ExponentialBackoffRetry;
+import org.corfudb.util.retry.IRetry;
+
+import java.lang.reflect.InvocationTargetException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+
+import static org.corfudb.runtime.view.TableRegistry.CORFU_SYSTEM_NAMESPACE;
+
+/**
+ * A client for log replication
+ */
+@Slf4j
+public class LogReplicationServiceClient {
+
+    public static final String LR_REGISTRY_TABLE_NAME = "LogReplicationRegistrationTable";
+    public static final String LR_SOURCE_METADATA_TABLE_NAME = "LogReplicationSourceMetadataTable";
+
+    /**
+     * Wait interval between consecutive sync attempts to cap exponential back-off
+     */
+    private static final int SYNC_THRESHOLD = 120;
+
+    private final CorfuStore corfuStore;
+    private final Table<ClientRegistrationId, ClientRegistrationInfo, Message> replicationRegistrationTable;
+    private final Table<ClientDestinationInfoKey, DestinationInfoVal, Message> sourceMetadataTable;
+
+    private final ClientRegistrationId clientKey;
+    private final ClientRegistrationInfo clientInfo;
+
+    private final String clientName;
+    private final String modelEnum;
+
+    /**
+     * Constructor for LogReplicationClient
+     *
+     * @param runtime Corfu Runtime
+     * @param clientName String representation of the client name
+     * @throws InvocationTargetException InvocationTargetException
+     * @throws NoSuchMethodException NoSuchMethodException
+     * @throws IllegalAccessException IllegalAccessException
+     */
+    public LogReplicationServiceClient(CorfuRuntime runtime, String clientName) throws InvocationTargetException, NoSuchMethodException, IllegalAccessException {
+        this.corfuStore = new CorfuStore(runtime);
+        this.clientName = clientName;
+
+        this.modelEnum = "LOGICAL_GROUPS";
+
+        this.clientKey = ClientRegistrationId.newBuilder()
+                .setClientName(clientName)
+                .build();
+        this.clientInfo = ClientRegistrationInfo.newBuilder()
+                .setClientName(clientName)
+                .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                .setRegistrationTime(Timestamp.newBuilder().setSeconds(new Date().getTime() / 1000).build())
+                .build();
+
+        this.replicationRegistrationTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, LR_REGISTRY_TABLE_NAME, ClientRegistrationId.class,
+                ClientRegistrationInfo.class, null, TableOptions.builder().build());
+        this.sourceMetadataTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, LR_SOURCE_METADATA_TABLE_NAME, ClientDestinationInfoKey.class,
+                DestinationInfoVal.class, null, TableOptions.builder().build());
+
```

Comment:
Actually the TableOptions field needs to be specified, normally we use the vClass to do that. Take this as an example:
https://github.com/CorfuDB/CorfuDB/blob/dcab1d42738ba183bdd04a5feffd6a3b9dad4594/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java#L84-L90